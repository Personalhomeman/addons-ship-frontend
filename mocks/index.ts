import { AppVersion } from '@/models';
import { TestDevice } from '@/models/test-device';

export const mockAppVersion: AppVersion = {
  id: 123,
  version: '1.0.3',
  platform: 'ios',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  minimumOs: '10.2',
  minimumSdk: 'dunno',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  bundleId: 'bundle-id',
  size: 4567034,
  supportedDeviceTypes: ['iphone', 'ipad'],
  distributionType: 'development',
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  promotionalText: 'Promotional Text',
  keywords: 'Keywords',
  reviewNotes: 'Review Notes',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  scheme: 'piramid',
  configuration: 'canary',
  publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
};

export const mockTestDevices: TestDevice[] = [
  { deviceId: 'some-device-id-01', deviceType: 'ios', owner: 'test-user-1' },
  { deviceId: 'some-device-id-02', deviceType: 'android', owner: 'test-user-1' },
  { deviceId: 'some-device-id-03', deviceType: 'ios', owner: 'test-user-2' }
];
