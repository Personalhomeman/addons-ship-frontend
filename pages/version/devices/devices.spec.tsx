import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import DevicesView from './view';
import { AppVersionDevices } from './';
import { mockTestDevices, mockApp } from '@/mocks';

describe('AppVersion', () => {
  it('renders the details tab correctly with UDID', () => {
    const tree = toJSON(shallow(<DevicesView projectType="ios" devices={mockTestDevices} />));
    expect(tree).toMatchSnapshot();
  });

  it('renders the details tab correctly with UUID', () => {
    const tree = toJSON(shallow(<DevicesView projectType="android" devices={mockTestDevices} />));
    expect(tree).toMatchSnapshot();
  });

  it('renders the details tab correctly with device ID', () => {
    const tree = toJSON(shallow(<DevicesView projectType="other" devices={mockTestDevices} />));
    expect(tree).toMatchSnapshot();
  });

  it('AppVersionDevices', () => {
    const tree = toJSON(shallow(<AppVersionDevices app={mockApp} testDevices={mockTestDevices} />));
    expect(tree).toMatchSnapshot();
  });
});
