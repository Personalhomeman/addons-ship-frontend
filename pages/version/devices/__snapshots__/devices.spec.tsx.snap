// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion AppVersionDevices 1`] = `
<_default
  devices={
    Array [
      Object {
        "deviceId": "some-device-id-01",
        "deviceType": "ios",
        "owner": "test-user-1",
      },
      Object {
        "deviceId": "some-device-id-02",
        "deviceType": "android",
        "owner": "test-user-1",
      },
      Object {
        "deviceId": "some-device-id-03",
        "deviceType": "ios",
        "owner": "test-user-2",
      },
    ]
  }
  projectType="other"
/>
`;

exports[`AppVersion renders the details tab correctly with UDID 1`] = `
<Table
  type="card"
>
  <TableHeader>
    <TableHeaderRow>
      <TableHeaderCell>
        User
      </TableHeaderCell>
      <TableHeaderCell>
        Device
      </TableHeaderCell>
      <TableHeaderCell>
        UDID
      </TableHeaderCell>
    </TableHeaderRow>
  </TableHeader>
  <TableBody>
    <TableRow
      key="some-device-id-01"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-01
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-02"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        android
      </TableCell>
      <TableCell>
        some-device-id-02
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-03"
    >
      <TableCell>
        test-user-2
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-03
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;

exports[`AppVersion renders the details tab correctly with UUID 1`] = `
<Table
  type="card"
>
  <TableHeader>
    <TableHeaderRow>
      <TableHeaderCell>
        User
      </TableHeaderCell>
      <TableHeaderCell>
        Device
      </TableHeaderCell>
      <TableHeaderCell>
        UUID
      </TableHeaderCell>
    </TableHeaderRow>
  </TableHeader>
  <TableBody>
    <TableRow
      key="some-device-id-01"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-01
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-02"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        android
      </TableCell>
      <TableCell>
        some-device-id-02
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-03"
    >
      <TableCell>
        test-user-2
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-03
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;

exports[`AppVersion renders the details tab correctly with device ID 1`] = `
<Table
  type="card"
>
  <TableHeader>
    <TableHeaderRow>
      <TableHeaderCell>
        User
      </TableHeaderCell>
      <TableHeaderCell>
        Device
      </TableHeaderCell>
      <TableHeaderCell>
        Device ID
      </TableHeaderCell>
    </TableHeaderRow>
  </TableHeader>
  <TableBody>
    <TableRow
      key="some-device-id-01"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-01
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-02"
    >
      <TableCell>
        test-user-1
      </TableCell>
      <TableCell>
        android
      </TableCell>
      <TableCell>
        some-device-id-02
      </TableCell>
    </TableRow>
    <TableRow
      key="some-device-id-03"
    >
      <TableCell>
        test-user-2
      </TableCell>
      <TableCell>
        ios
      </TableCell>
      <TableCell>
        some-device-id-03
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
